### ISSUE TITLE
Cannot remove directories from adopted backup

### ISSUE THREAD

**Issue by dontask2035** [jlesage/docker-crashplan-pro#231]
Hello, 

I have Crashplan working on Docker. Thanks Jlesage!  

My backup on Crashplan has 3tb. I don't need that much storage and is backing up old data. I can login to the web version of crashplan and see my old drives that I no longer need to back up. I would like to remove them from my backup set. However, crashplan only allows this to be done from the device that it is currently backed up. 

On my docker/crashplan/jlesage crashplan client, I can go into manage files and see the drive letters from previous backups, but there are no folders. For example, on previous drive D, there is a folder named DUPLICATE that I would like to unselect from my backup set, but that folder does not appear on my docker crashplan pro client. Any ideas? 

Secondly, when I mapped my folders for crashplan, I mapped them based on the name of the folders on the NAS.  I.e., Volume1/Data/ became /Data. The backup on crashplan has this as /root on the manage files, and on the actual server is /Data. I have no issues with this, but the install help files say that this should be at /storage and if there is a misconfiguration I would rather find it now than after a week of uploading. I presume that mapping my drives as I have is not a problem? 

Thanks in advance.

**Comment by jvarrone** [comment]
I'm having a similar problem with the CP client adding /root before volume1. So it appears in the client as /root/volume1 which doesn't correspond to how CP Central has stored the backups as simply /volume1  How do you do the mapping? I'm at a loss as to where to find the mapping in the Gui interface.

**Comment by jlesage** [comment]
> On my docker/crashplan/jlesage crashplan client, I can go into manage files and see the drive letters from previous backups, but there are no folders. For example, on previous drive D, there is a folder named DUPLICATE that I would like to unselect from my backup set, but that folder does not appear on my docker crashplan pro client. Any ideas?

Can you deselect the drive letter itself ?  Also, in the configuration of your backup, you can set the amount of time CP keep deleted files.  By default it's 1 year.  You can reduce the value if it's too much for you.

> Secondly, when I mapped my folders for crashplan, I mapped them based on the name of the folders on the NAS. I.e., Volume1/Data/ became /Data. The backup on crashplan has this as /root on the manage files, and on the actual server is /Data. I have no issues with this, but the install help files say that this should be at /storage and if there is a misconfiguration I would rather find it now than after a week of uploading. I presume that mapping my drives as I have is not a problem?

This is not a problem.  It's not required to use `/storage`.

**Comment by dontask2035** [comment]
Yes, I can deselect the entire volume, but there is data on the volume that I would like to keep. For example, there is C, D and E from an adopted Win10 backup. There is data on D that I would like to remove from the backup, but the client on Docker only shows C D and E with check marks. If I click through to the drive, there is nothing. 

![image](https://user-images.githubusercontent.com/32400598/77777045-2f144f00-700c-11ea-9814-f717f8db1b6f.png)

**Comment by jlesage** [comment]
There are some limitation when adopting backup of a different OS.  Maybe this is one of them...
See : https://support.code42.com/CrashPlan/6/Configuring/Replace_a_device_with_a_different_operating_system_or_file_system

**Comment by dontask2035** [comment]
That does not appear to be the issue. Those support documents state that the structure may be different after changing systems. However, in that scenario the old data is still in the structure, just not aligned with the new structure. That did happen here. Moving from a Win10 system, the old drives are C D and E, whereas the new backup is in Root. But either way, I should be able to deselect old backup folders.

**Comment by jlesage** [comment]
But the document also says (look at the Mac to Windows example):

> You can only view and manage the top-level folders included in your file selection from a previous device.

**Comment by johnripper1** [comment]
If you map the drives with other paths than before I suggest not to remove the old ones, as the version and deleted files will then be deleted as well. This means you lose the version history and the deleted file history.

I keep the old pathes since years.

Not sure if you want that.

**Comment by jvarrone** [comment]
I'm having the problem with the CP client adding /root before volume1. So it appears in the client as /root/volume1 which doesn't correspond to how CP Central has stored the backups as simply /volume1 How do you do the mapping? I'm at a loss as to where to find the mapping in the Gui interface. Can anyone give me exact instructions on what to do to get reconnected? I've tried everything I can think to fix it from these threads but nothing is working. The client always shows /root/volume1. How do I get rid of the "root" prefix? I would even pay someone to help me fix this. CP support won't touch it because they keep repeating - we don't support a Docker install.

**Comment by SJLBoulder** [comment]
jvarrone - I believe that is normal.  That's how mine looks.  Why is that a problem?  I've successfully restored files to their original location.  So, I consider this working as designed.

**Comment by jvarrone** [comment]
The files are showing as deleted at CP Central. (I have retention set to never remove deleted files so I can get to them) When I try to set up CP in Docker, I can't get to Volume1 to reconnect them. The CP interface show /root/Volume1 which doesn't exist at CP Central. Cp Central has them stored simply as /Volume1/foldernames 
Am I making sense?
![cpcentral](https://user-images.githubusercontent.com/3275462/81316138-31c98180-9059-11ea-90c7-f7615a03ef26.JPG)

**Comment by jlesage** [comment]
This all depends on how you mapped your folders to the container.  If, inside the container, your data is under `/root/volume1`, then the same thing will be reflected on CP side.

**Comment by madfusker** [comment]
What does your docker setup look like?  In the end it doesn't seem like it has to match, but, if you do a direct restore from the CP app to "original location", you may want a full system path to put it back into the right spot.  Here is mine, but I run the access to shares in ro mode unless I need to restore (then I change it to rw):

> docker run --restart unless-stopped -d \
>     --name=crashplan-pro \
> 	--restart=always \
>     -e USER_ID=0 \
>     -e GROUP_ID=0 \
>     -p 5800:5800 \
>     -p 5900:5900 \
>     -e TZ=America/Chicago \
>     -e CRASHPLAN_SRV_MAX_MEM=4096M \
>     -v /share/docker/appdata/crashplan-pro:/config:rw \
>     -v /share/CACHEDEV1_DATA:/share:ro \
>     jlesage/crashplan-pro

Note that I used the full path of each location.  Also see below that QNAP on my TS877 has the share symlinked to the raw device location CACHEDEV1.  I do this intentionally in case QNAP changes the disk device name, or I put the archive on a different device, it's still going to be /share/Pictures in the symlinked path.
> 
> # ls -l /share/Pictures
> lrwxrwxrwx 1 admin administrators 23 2020-05-09 21:48 /share/Pictures -> CACHEDEV1_DATA/Pictures/

Lastly, with the data dedupe feature on CP, you should be ok to change your path now. even a week after backing up.  Simply set the new path and let CP run.  It shouldn't have to backup files it has already sent up.  I had this happen when I moved mine from a TS-569L to a TS-877 which was also legacy volume to new style volume.  If I look in crashplan cloud today, I can still see a **MD0**/Pictures, AND my new path **CACHEDEV1_DATA**/Pictures.   I just leave it in the cloud as-is since there are version still stored there from the old NAS in MD0.  Hope that helps.

-sker

**Comment by madfusker** [comment]
So my app has Root > share >Pictures

![Capture](https://user-images.githubusercontent.com/39440750/81505593-042d4400-92b6-11ea-9d03-036148a6d5af.JPG)

**Comment by madfusker** [comment]
I see now there are multiple threads of this, and looks like it was already answered in the other thread...
